src/scicookie/{{cookiecutter.project_slug}}/.makim.yaml

```diff
@@ -17,18 +22,86 @@ groups:
           rm -f .coverage
           rm -fr htmlcov/
           rm -fr .pytest_cache
+          rm -fr .mypy_cache
+          rm -fr .ruff_cache
 
   docs:
     targets:
       build:
         help: Build documentation
         run: |
+          {%- if cookiecutter.documentation_engine == "jupyter-book" %}
           rm -rf docs/_build
           jupyter-book build docs
+          {%- elif cookiecutter.documentation_engine == "mkdocs" %}
+          mkdocs build --config-file docs/mkdocs.yaml
+          {%- elif cookiecutter.documentation_engine == "sphinx" %}
+          sphinx-apidoc -o docs/_build {{ package_path }}
+          {%- endif %}
 
       preview:
         help: Preview documentation page locally
-        run: echo "[EE] Not implemented yet!"
+        dependencies:
+          - target: docs.build
+        run: |
+          {%- if cookiecutter.documentation_engine == "jupyter-book" %}
+          cd docs/_build/html/ && python -m http.server
+          {%- elif cookiecutter.documentation_engine == "mkdocs" %}
+          mkdocs build --config-file docs/mkdocs.yaml
+          {%- elif cookiecutter.documentation_engine == "sphinx" %}
+          cd docs/_build && python -m http.server
+          {%- endif %}
+
+  tests:
+    targets:
+      linter:
+        help: Run linter tools
+        run: |
+          pre-commit install
```

Comment:
yep, the idea is to force the user to install the pre-commit hook locally ... maybe it is not the best approach, I am currently investigating some better approach for doing that.

for poetry I found this: https://pypi.org/project/poetry-pre-commit-plugin/
but it seems it is not mature yet (all its tests are not executed: https://github.com/vstrimaitis/poetry-pre-commit-plugin/blob/master/tests/test_plugin.py

maybe we could do it as a post-gen step .. so it could work for all the projects .. but we would need to create the environment first ... so not sure if it is a good idea or not

nodejs projects are very nice in this topic .. because when you install it for development, it automatically installs pre-commit hook.

so if it is not a problem, I would keep it for now, and in the meantime I will try to figure out a good approach that would works for all the build-systems